### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by beteland123**
in this branch I:
- Add association for the model of the post, comment, user, and like
- Add methods for post counter, likes counter, comment counter, recent posts and recent comment 
- Add data to the databe

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
+  after_create :update_post_likes_counter
+  before_destroy :update_likes_before_destroy
+
+  def update_post_likes_counter
+    post.increment!(:likescounter)
+  end
+
+  def update_likes_before_destroy
+    post.decrement!(:likescounter)
+  end
```
Comment (levy002) [inline app/models/like.rb:16]: - [ ] In your `Like` model, consider making the `update_post_likes_counter` and `update_likes_before_destroy` methods private. By marking them as private, you signify that these methods are meant for internal use within the `Like` model and should not be accessed or called from outside the class. This encapsulation helps maintain clean and organized code, preventing unintended usage and potential issues in the future.

```suggestion
private

  def update_post_likes_counter
    post.increment!(:likescounter)
  end

  def update_likes_before_destroy
    post.decrement!(:likescounter)
  end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  def increment_user_posts_count
+    author.increment!(:postcounter)
+  end
+
+  def update_post_counter_before_destroy
+    author.decrement!(:postcounter)
+  end
```
Comment (levy002) [inline app/models/post.rb:21]: - [ ] In your `Post` model, consider making the `increment_user_posts_count` and `update_post_counter_before_destroy` methods private. By marking them as private, you signify that these methods are meant for internal use within the `Post` model and should not be accessed or called from outside the class. This encapsulation helps maintain clean and organized code, preventing unintended usage and potential issues in the future.

```suggestion
private 

  def increment_user_posts_count
    author.increment!(:postcounter)
  end

  def update_post_counter_before_destroy
    author.decrement!(:postcounter)
  end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,15 @@
...
+  after_create :update_post_comments_counter
+  before_destroy :update_comments_before_destroy
+
+  def update_post_comments_counter
+    post.increment!(:commentscounter)
+  end
+
+  def update_comments_before_destroy
+    post.decrement!(:commentscounter)
+  end
```
Comment (levy002) [inline app/models/comment.rb:16]: - [ ] In your `Comment` model, consider making the `update_post_comments_counter` and `update_comments_before_destroy` methods private. By marking them as private, you signify that these methods are meant for internal use within the `Comment` model and should not be accessed or called from outside the class. This encapsulation helps maintain clean and organized code, preventing unintended usage and potential issues in the future.

```suggestion
private

  def update_post_comments_counter
    post.increment!(:commentscounter)
  end

  def update_comments_before_destroy
    post.decrement!(:commentscounter)
  end
```

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @beteland123   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @beteland123 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**